# WACC

In [12]:
def calculate_wacc():
    # Aqui calculo el wacc, con los datos del problema
    Rf = 0.035  # Tasa libre de riesgo (3.5%)
    beta = 2.14  # Beta
    market_risk_premium = 0.0435  # Prima del mercado (4.35%)
    cost_of_debt = 0.036  # Costo de la deuda (3.6%)
    tax_rate = 0.30  # Tipo impositivo (30%)
    
    # Valores de Mercado
    equity_value = 1_000_000 * 1.22  # Patrimonio (1 millón de acciones * $1.22/Accion)
    debt_value = 6_300_000 + 4_500_000  # Deuda total (L/P + C/P)
    total_value = equity_value + debt_value  # Valor del patrimonio neto 

    # Costo de financiación usando CAPM
    cost_of_equity = Rf + beta * market_risk_premium

    # Formula del WACC
    wacc = (equity_value / total_value) * cost_of_equity + (debt_value / total_value) * cost_of_debt * (1 - tax_rate)
    
    return wacc, cost_of_equity

print(f"WACC: {wacc:,.2f}") # Obtenemos el WACC
print (f"Costo de Financiación {cost_of_equity:,.2f}") # Obtenemos el costo de financiación 

WACC: 0.04
Costo de Financiación 0.13


# DCF

In [10]:
def dcf_valuation():
    # Datos del Ejercicio
    sales = [42_500_000]  # Ventas Iniciales
    sales_growth = [0.06, 0.06, 0.05, 0.04, 0.04]  # Incremento de las ventas en 5 años
    gross_margin = [0.52, 0.52, 0.52, 0.45, 0.45]  # Incremento del margen bruto
    personnel_reduction = 700_000  # Reducción en los costos de mano de obra
    personnel_growth = 0.03  # Incremento del costo de trabajo después del tercer año
    capex_percent = [0.07, 0.07, 0.06, 0.06, 0.06]  # CAPEX en % de ventas
    working_capital_percent = 0.15  # Estimativa de cambio en el working capital
    wacc, _ = calculate_wacc()  # WACC obtenido antes
    terminal_growth = 0.015  # Ritmo de crecimiento de los perpetuos

    # Proyección de ventas y EBITDA
    for i in range(1, 6):
        new_sales = sales[-1] * (1 + sales_growth[i-1])
        sales.append(new_sales)

    # Computo EBITDA, CAPEX, y flujos de caja libre
    free_cash_flows = []
    for i in range(5):
        ebitda = sales[i] * gross_margin[i]
        capex = sales[i] * capex_percent[i]
        free_cash_flow = ebitda - capex  
        free_cash_flows.append(free_cash_flow)

    # Calculo los perpetuos
    terminal_value = free_cash_flows[-1] * (1 + terminal_growth) / (wacc - terminal_growth)

    # Calculato el valor presente de los flujos de caja futuros
    npv = sum(fcf / ((1 + wacc) ** (i + 1)) for i, fcf in enumerate(free_cash_flows))
    npv_terminal_value = terminal_value / ((1 + wacc) ** 5)

    # Valor de Negocio es equivalente al Valor presente de los FCF + Valor Presente de los perpetuos
    business_value = npv + npv_terminal_value

    return business_value

business_value = dcf_valuation()
print(f"Valor de Negocio: ${business_value:,.2f}") # Obtenemos el valor del negocio


Valor de Negocio: $930,534,884.04


# Enunciado

**Calcular el Coste de Capital (WACC)**

El **WACC** ya ha sido calculado y es:

$$
WACC = 0.0356 = 3.56\%
$$


**Valoración del Target por el Método DCF**

La **Valoración del Negocio** se realiza usando el método de **Flujo de Caja Descontado (DCF)**. El valor total del negocio fue calculado:

$$
\text{Valor del negocio} = \$930,534,884.04
$$

Este valor refleja la suma de los flujos de caja proyectados para los próximos años, descontados al presente utilizando el WACC, más el valor terminal (valor a perpetuidad).

#### Valor de las Acciones:
  

$$
\text{Valor de las acciones} = \frac{930,534,884.04 - 10,800,000}{1,000,000} = \$919.73 \, \text{por acción}
$$

El nuevo valor de las acciones después de la evaluación DCF con sinergias sería de aproximadamente **$919.73 por acción**, comparado con el valor actual de **$1.22 por acción** en el mercado.

**¿Deben Tenerse en Cuenta las Potenciales Sinergias en los Flujos de Caja?**

**Sí**: Las sinergias representan mejoras en la eficiencia operativa o financiera como resultado de la fusión o adquisición. Estas sinergias pueden reducir los costos operativos, mejorar las ventas, o hacer que el uso de los activos sea más eficiente, lo que a su vez incrementa el valor futuro del negocio. En el caso del target, las sinergias se traducen en una reducción de costos (como personal, alquiler, servicios, etc.), lo que mejora el flujo de caja y, en última instancia, el valor empresarial.

**¿Es Adecuada la Tasa de Crecimiento Perpetuo?**

Esta tasa está en línea con el crecimiento esperado de la economía a largo plazo. Una tasa de crecimiento perpetuo que exceda este valor podría ser demasiado optimista, especialmente para una empresa en un mercado competitivo y maduro como el de la robótica. Además, una tasa del 1.5% se encuentra ligeramente por encima de la inflación (sin considerar los escenarios recientes), lo que es razonable para mantener el valor del dinero en el tiempo.

**Máxima Tasa Sugerida**: 
- La máxima tasa de crecimiento perpetuo que sugeriría sería alrededor del **2%**. Esto estaría alineado con un crecimiento económico sostenido sin ser demasiado optimista ni poner en riesgo la evaluación con un crecimiento perpetuo irreal. Como expliqué en la anterior actividad, la tasa se alinea con el modelo de crecimiento de Solow, para economias maduras de largo plazo.

**¿Son Suficientes las Inversiones en CAPEX para el Cálculo de la Perpetuidad?**

Las inversiones en **CAPEX del 7% y luego 6% de las ventas** parecen suficientes para los primeros años, pero pueden no ser adecuadas para la perpetuidad.
- En las primeras fases, el 7% y 6% de CAPEX en relación con las ventas es apropiado para una empresa que está experimentando crecimiento. Esto asegura que la empresa está invirtiendo lo suficiente en infraestructura y tecnología para mantener su posición competitiva. En el sector de robótica, si hablamos de innovación, es natural que los pioneros desfruten de crecimiento acelerado mientras la ola de replicadores se empieza a configurar, como detallado por Schumpeter.
  
- **Para la perpetuidad**, es probable que la empresa, una vez madura, no necesite invertir tanto en CAPEX. Una tasa de CAPEX más baja en la perpetuidad (por ejemplo, 3%-4%) sería más razonable, ya que las empresas maduras tienden a reducir sus necesidades de inversión en activos fijos y enfocarse en el mantenimiento de su capacidad actual.
